include/ceed/ceed.h

```diff
@@ -473,6 +473,14 @@ CEED_EXTERN int  CeedOperatorApplyAddActive(CeedOperator op, CeedVector in, Ceed
 CEED_EXTERN int  CeedOperatorAssemblyDataStrip(CeedOperator op);
 CEED_EXTERN int  CeedOperatorDestroy(CeedOperator *op);
 
+// Compatibility with previous composite CeedOperator naming
+#define CeedCompositeOperatorCreate(a, b) CeedOperatorCreateComposite(a, b)
+#define CeedCompositeOperatorAddSub(a, b) CeedOperatorCompositeAddSub(a, b)
+#define CeedCompositeOperatorGetNumSub(a, b) CeedOperatorCompositeGetNumSub(a, b)
+#define CeedCompositeOperatorGetSubList(a, b) CeedOperatorCompositeGetSubList(a, b)
+#define CeedCompositeOperatorGetSubByName(a, b) CeedOperatorCompositeGetSubByName(a, b, c)
+#define CeedCompositeOperatorGetMultiplicity(a, b, c, d) CeedOperatorCompositeGetMultiplicity(a, b, c, d)
```

Comment:
We could also put deprecation warning attribute on these too? Would require changing to actual function definitions rather than macro functions.

Reply:
It gives better error messages anyway if these are replaced. They could be `static inline` unless they might need to be real symbols (callable via FFI or without recompiling source).

Reply:
see the latest commit

Reply:
It is, but the format pre-C23 format was different (e.g. `__attribute__ ((deprecated))` for GCC). PETSc should have a list of them for their deprecation macros.

Reply:
Is there a downside to using the `__attribute__` syntax unless `__STDC_VERSION__ >= 202311L`?

Reply:
Not that I know? Hopefully this isn't something that we'll need much after the 1.0 release